state-chain/runtime/src/chainflip/bitcoin_elections.rs

```diff
@@ -302,11 +311,32 @@ impl Hook<btc::BlockNumber, Vec<DepositChannelDetails<Runtime, BitcoinInstance>>
 	}
 }
 
+pub type BitcoinLiveness = Liveness<
+	BlockNumber,
+	Hash,
+	cf_primitives::BlockNumber,
+	OnCheckCompleteHook,
+	<Runtime as Chainflip>::ValidatorId,
+>;
+
+pub struct OnCheckCompleteHook;
+
+impl OnCheckComplete<<Runtime as Chainflip>::ValidatorId> for OnCheckCompleteHook {
+	fn on_check_complete(validator_ids: BTreeSet<<Runtime as Chainflip>::ValidatorId>) {
+		<Reputation as OffenceReporter>::report_many(
+			Offence::FailedLivenessCheck(ForeignChain::Bitcoin),
+			validator_ids,
+		);
+	}
+}
```

Comment:
we can probably macrofy this out with the solana one - we're going to need it for all the other chains too